bigtable/beam/bulk-data-generator/src/test/java/bigtable/BulkWriteTest.java

```diff
@@ -0,0 +1,115 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package bigtable;
+
+import static org.junit.Assert.assertNotNull;
+
+import com.google.cloud.bigtable.admin.v2.BigtableInstanceAdminClient;
+import com.google.cloud.bigtable.admin.v2.models.CreateInstanceRequest;
+import com.google.cloud.bigtable.admin.v2.models.Instance;
+import com.google.cloud.bigtable.admin.v2.models.StorageType;
+import com.google.common.truth.Truth;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.UUID;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BulkWriteTest {
+
+  private static final String PROJECT_ENV = "GOOGLE_CLOUD_PROJECT";
+  private static final String INSTANCE_ID = "ins-" + UUID.randomUUID().toString().substring(0, 10);
+  private static final String CLUSTER_ID = "cl-" + UUID.randomUUID().toString().substring(0, 10);
+  private static final String REGION_ID = "us-central1";
+  private static final String ZONE_ID = "us-central1-b";
+  private static final int NUM_TABLES_TO_CREATE = 3;
+  private static final double TABLE_SIZE = .5;
+  private static final double BIGTABLE_SIZE = TABLE_SIZE * NUM_TABLES_TO_CREATE;
+
+  private static String projectId;
+  private ByteArrayOutputStream bout;
+
+  private static String requireEnv(String varName) {
+    String value = System.getenv(varName);
+    assertNotNull(
+        String.format("Environment variable '%s' is required to perform these tests.", varName),
+        value);
+    return value;
+  }
+
+  @BeforeClass
+  public static void beforeClass() {
+
+    projectId = requireEnv(PROJECT_ENV);
+    try (BigtableInstanceAdminClient instanceAdmin =
+        BigtableInstanceAdminClient.create(projectId)) {
+      CreateInstanceRequest request = CreateInstanceRequest.of(INSTANCE_ID)
+          .addCluster(CLUSTER_ID, ZONE_ID, 1, StorageType.SSD);
+      Instance instance = instanceAdmin.createInstance(request);
+    } catch (IOException e) {
+      System.out.println("Error during BeforeClass while creating instance: \n" + e.toString());
+      Assert.fail();
+    }
+  }
+
+  @Before
+  public void setupStream() {
+    bout = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(bout));
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    try (BigtableInstanceAdminClient instanceAdmin =
+        BigtableInstanceAdminClient.create(projectId)) {
+      instanceAdmin.deleteInstance(INSTANCE_ID);
+    } catch (IOException e) {
+      System.out.println("Error during AfterClass while deleting instance: \n" + e.toString());
+    }
+  }
+
+  @Test
+  public void testBulkWrite() {
+    BulkWrite.main(
```

Comment:
As I mentioned in a previous comment, I would consider rewriting this test so that you don't need to pass the test variables into your test as args to your `main()` method.

Here's an example of a test from the Vertex AI samples:
https://github.com/googleapis/java-aiplatform/blob/master/samples/snippets/src/test/java/aiplatform/CreateTrainingPipelineImageObjectDetectionSampleTest.java#L99-L101

Reply:
done